### ISSUE TITLE
consider annotating dependency information

### ISSUE THREAD

**Issue by kevinushey** [rstudio/htmltools#59]
It would be useful if [`renderDependencies()`](https://github.com/rstudio/htmltools/blob/master/R/html_dependency.R#L375-L464) annotated the injected `<script>` / `<link>` tags with data attributes describing the source of each dependency, e.g.

```
 <script
        data-name="htmlwidgets"
        data-version="0.6"
        data-script="htmlwidgets.js"
        data-stylesheet=""
        src="/path/to/R/library/htmlwidgets/www/htmlwidgets.js"></script>
```

This would make it possible to recover the source of a dependency after e.g. `rmarkdown::render()` is done with the document.

**Comment by jcheng5** [comment]
I'm a little leery of putting the user library's absolute path in the html output, if the latter may be published somewhere. Can this be constrained to cases where we know the output isn't going anywhere?

**Comment by kevinushey** [comment]
Sorry, I should have been more clear -- all I'm advocating for is the inclusion of the `data-*` attributes, which should be easy to draw from the dependency information already passed into `renderDependencies()`. (The `src` field would just be whatever is normally passed in)

**Comment by yihui** [comment]
Actually I wished the dependency object had a `package` attribute, and `src` could be resolved at a later time using `system.file(src, package = package)` (if the `package` name is provided). This will hide the absolute path, and make the dependencies more portable since they will be independent of the R library path.

**Comment by jjallaire** [comment]
I agree. I think this could be pretty seamlessly added (if there is a
`package` attribute then `src` can be relative to the package).
